### PR TITLE
fix: Soft navigation bug fixes and new soft navigation tests

### PR DESCRIPTION
**Pull request by ellisong**
Fixes several issues tied to the soft navigations feature.  Calling newrelic.interaction() prior to a page load now creates a separate "api" trigger node. Multiple interactions managed simultaneously now have correct timings. Fixes a bug in soft navigation where API-triggered interactions fail to close under default heuristic (URL + DOM change) because the mutation observer is not started. Finally, 'popstate' events are now valid interaction triggers for soft navigations.
---

### Overview

Fixes a few bugs related to the soft navigation feature and adds new tests ported from the old spa feature. Not all of the tests are portable due to functional differences or limitations.

### Related Issue(s)

https://new-relic.atlassian.net/browse/NR-300966

bug: https://new-relic.atlassian.net/browse/NR-343675

### Testing

Make sure all tests pass on all browsers.

### CODE REVIEW

File: src/features/soft_navigations/aggregate/interaction.js [hunk 1]
```diff
@@ -105,7 +105,7 @@ export class Interaction extends BelNode {
    */
   isActiveDuring (timestamp) {
     if (this.status === INTERACTION_STATUS.IP) return this.start <= timestamp
-    return (this.status === INTERACTION_STATUS.FIN && this.start <= timestamp && this.end >= timestamp)
+    return (this.status === INTERACTION_STATUS.FIN && this.end > timestamp && this.start <= timestamp)
```
[the remark this answers is not in the document]
Comment (ellisong) [reply src/features/soft_navigations/aggregate/interaction.js:108]: I think was a change I accidentally left in. I'll revert it.

File: src/features/soft_navigations/instrument/index.js [hunk 2]
```diff
@@ -29,22 +28,22 @@ export class Instrument extends InstrumentBase {
       }, true)
     })
 
-    const trackURLChange = () => handle('newURL', [now(), '' + window.location], undefined, this.featureName, this.ee)
+    const trackURLChange = () => handle('newURL', [performance.now(), '' + window.location], undefined, this.featureName, this.ee)
```
[the remark this answers is not in the document]
Comment (ellisong) [reply src/features/soft_navigations/instrument/index.js:31]: I think I was trying to get the tests to pass. I'll revert change and see if it tests still pass.

File: src/features/soft_navigations/aggregate/index.js [hunk 3]
```diff
@@ -68,21 +75,23 @@ export class Aggregate extends AggregateBase {
...
     if (this.interactionInProgress?.createdByApi) return // api-started interactions cannot be disrupted aka cancelled by UI events (and the vice versa applies as well)
-    if (this.interactionInProgress?.done() === false) return
+    if (this.interactionInProgress?.done() === false) return // current in-progress is blocked from closing, e.g. by 'waitForEnd' api option
+
+    const oldURL = eventName === INTERACTION_TRIGGERS[3] ? this.latestHistoryUrl : undefined // see related comment in 'newURL' handler above, 'popstate'
+    this.interactionInProgress = new Interaction(this.agentIdentifier, eventName, startedAt, this.latestRouteSetByApi, oldURL)
 
-    this.interactionInProgress = new Interaction(this.agentIdentifier, eventName, startedAt, this.latestRouteSetByApi)
-    if (eventName === 'click') {
+    if (eventName === INTERACTION_TRIGGERS[0]) { // 'click'
```
[the remark this answers is not in the document]
Comment (ellisong) [reply src/features/soft_navigations/aggregate/index.js:94]: Yeah, I can refactor it to an object

File: src/features/soft_navigations/aggregate/interaction.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ellisong) [reply src/features/soft_navigations/aggregate/interaction.js:108]: I tried reverting it but the `should set correct start and end values on multiple custom interactions` is not passing without the `this.end > timestamp` change after multiple attempts.

**Review by ptang-nr — APPROVED** [review]

**Pull request merged**